src/game_interpreter.cpp

```diff
@@ -4626,6 +4663,340 @@ bool Game_Interpreter::CommandManiacSetGameOption(lcf::rpg::EventCommand const&)
 	return true;
 }
 
+bool Game_Interpreter::CommandManiacControlStrings(lcf::rpg::EventCommand const& com) {
+	if (!Player::IsPatchManiac()) {
+		return true;
+	}
+	//*parameter 0 - Modifier of the operation members
+	//	-bits 0..3:
+	//		Refers to the String mode: t[x], t[x..y], t[v[x]], t[v[x]..v[y]]
+	//	-bits 4..7:
+	//		Refers to the first argument, which can change depending on the type of operation:  x, v[x], v[v[x]] | string, t[x], t[v[x]]
+	//	-bits 8..11:
+	//		Refers to the second argument, which can change depending on the type of operation: x, v[x], v[v[x]] | string, t[x], t[v[x]]
+	//	-bits 12..15:
+	//		Refers to the third argument, which can change depending on the type of operation:  x, v[x], v[v[x]] | string, t[x], t[v[x]]
+	//	-bits 16..19:
+	//		Refers to the fourth argument, which can change depending on the type of operation: x, v[x], v[v[x]] | string, t[x], t[v[x]] (edge case for exMatch)
+	// 
+	//*parameter 1 - String index 0
+	// 
+	//*parameter 2 - String index 1 (optional for ranges)
+	// 
+	//*parameter 3 - general flags
+	//	-byte0:
+	//		Refers to the type of operation: asg, cat, toNum...
+	//	-byte1:
+	//		It is a flag that indicates sub-operation: actor[x].name, .actor[x].desc, ins, rep, subs, join... Used only in asg and cat operations
+	//	-byte2:
+	//		Flags, such as: extract, hex... There is also an edge case where the last argument of exRep is here
+	// 
+	//*parameters 4..n - arguments
+	int string_mode = com.parameters[0] & 15;
+	int string_id_0 = com.parameters[1];
+	int string_id_1 = com.parameters[2]; //for ranges
+
+	int is_range = string_mode & 1;
+
+	if (string_mode >= 2) {
+		string_id_0 = Main_Data::game_variables->Get(string_id_0);
+	}
+	if (string_mode == 3) {
+		string_id_1 = Main_Data::game_variables->Get(string_id_1);
+	}
+
+	int op    = (com.parameters[3] >>  0) & 255;
+	int fn    = (com.parameters[3] >>  8) & 255;
+	int flags = (com.parameters[3] >> 16) & 255;
+
+	int hex_flag     = (flags >> 1) & 1;
+	int extract_flag = (flags >> 2) & 1;
+	int first_flag = (flags >> 3) & 1;
```

Comment:
So stupid that this is a new command so there was full freedom of implementation and it still contains all these stupid bit-masking shit :facepalm: